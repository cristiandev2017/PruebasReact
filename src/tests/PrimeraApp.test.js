//import {render} from '@testing-library/react';
import React from 'react';
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en <PrimeraApp />", () => {
/*Esta seria sin usar enzyme    
test('debe de mostrar el mensaje "Hola, soy Cristian" ', () => { 
const saludo = "Hola, Soy Cristian";
//Aca renderizare el componente
const {getByText} = render(<PrimeraApp saludo={saludo}/>);
expect(getByText(saludo)).toBeInTheDocument();
})
*/

  test("Debe de mostrar <PrimeraApp /> Correctamente", () => {
    const saludo = "Hola, Soy Cristian";
    //shallow es parecido al render pero con otras propiedades
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    //Ahora el Assert
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el subtitulo enviado por props", () =>{
    //Arrange
    const saludo = "Hola, Soy Cristian";
    const subtitulo = "Soy un subtitulo";
    //Act
    const wrapper = shallow(  
      <PrimeraApp 
        saludo={saludo}
        subtitulo={subtitulo}
      />
    );
    //Asserts
    const textoParrafo = wrapper.find('p').text();
    expect(textoParrafo).toBe(subtitulo);
  })
});
