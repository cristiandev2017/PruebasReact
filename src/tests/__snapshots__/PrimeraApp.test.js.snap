// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> Correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Cristian
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
